cmd/ethrex/utils.rs

```diff
@@ -81,9 +82,20 @@ pub fn read_block_file(block_file_path: &str) -> Block {
         .unwrap_or_else(|_| panic!("Failed to decode block file {}", block_file_path))
 }
 
-pub fn read_genesis_file(genesis_file_path: &Path) -> Genesis {
+pub fn read_genesis_file(genesis_file_path: &Path) -> Result<Genesis, ChainError> {
     let genesis_file = std::fs::File::open(genesis_file_path).expect("Failed to open genesis file");
-    decode::genesis_file(genesis_file).expect("Failed to decode genesis file")
+    let genesis = decode::genesis_file(genesis_file).expect("Failed to decode genesis file");
+    // returns Paris by default if no newer fork is active
+    let fork = genesis.config.get_fork(genesis.timestamp);
```

Comment:
the purpose of this function is to get the current fork _after_ validating the genesis, not during/for validation.